src/main/kotlin/bajo/software/com/feature/validators/domain/IsValidName.kt

```diff
@@ -0,0 +1,19 @@
+package bajo.software.com.feature.validators.domain
+
+class IsValidName(
+    private val rules: NameValidatorRules,
+    private val nameValidatorSource: NameValidatorSource
+) {
+    suspend fun invoke(name: String): Boolean {
+        if(name.length < rules.MIN_LENGTH){
+            throw NameEmptyException()
+        } else if (name.length > rules.MAX_LENGTH){
+            throw NameExceedsLimitException()
+        } else {
+            return nameValidatorSource.getAll(name).isEmpty()
+        }
+    }
+}
+
+class NameEmptyException : IllegalArgumentException("The name need two characters at least")
+class NameExceedsLimitException : IllegalArgumentException("The name must be a maximum of 50 characters")
```

Comment:
```suggestion
data object NameEmptyException : IllegalArgumentException("The name need two characters at least")
data object NameExceedsLimitException : IllegalArgumentException("The name must be a maximum of 50 characters")
```

En caso de que no las juntes en una excepción, estos pueden ser objects :D

Reply:
Se cambian por enum class